biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/DownloadChemCompProvider.java

```diff
@@ -364,29 +359,20 @@ private static boolean downloadChemCompRecord(String recordName) {
 
 			HttpURLConnection uconn = HTTPConnectionTools.openHttpURLConnection(url);
 
-			InputStream conn = uconn.getInputStream();
-
-			FileOutputStream outPut = new FileOutputStream(localName);
-
-			GZIPOutputStream gzOutPut = new GZIPOutputStream(outPut);
+			try( PrintWriter pw = new PrintWriter(new GZIPOutputStream(new FileOutputStream(localName)));
+					BufferedReader fileBuffer = new BufferedReader(new InputStreamReader(uconn.getInputStream()));
+					) {
```

Comment:
In this case I think not, because `openHttpURLConnection` can throw the exceptions as well. I wanted the try-with to surround only the stream code.